api/v1alpha4/index/index.go

```diff
@@ -59,40 +67,20 @@ func indexMachineByNodeName(o client.Object) []string {
 	return nil
 }
 
-// IndexNodeByProviderID contains the logic to index Nodes by ProviderID.
-func IndexNodeByProviderID(o client.Object) []string {
-	node, ok := o.(*corev1.Node)
-	if !ok {
-		panic(fmt.Sprintf("Expected a Node but got a %T", o))
-	}
-
-	if node.Spec.ProviderID == "" {
-		return nil
-	}
-
-	providerID, err := NewProviderID(node.Spec.ProviderID)
-	if err != nil {
-		// Failed to create providerID, skipping.
-		return nil
-	}
-	return []string{providerID.IndexKey()}
-}
-
 // AddMachineProviderIDIndex adds the machine providerID index to the
 // managers cache.
 func AddMachineProviderIDIndex(ctx context.Context, mgr ctrl.Manager) error {
```

Comment:
Now they are in a separated package, what about having each index in a separated file?

Reply:
+1, maybe split them by type?

Reply:
@vincepri so indexes on `clusterv1.Machine` and `corev1.Node` in separate files? (`machine.go` and `node.go`?)